packages/vaadin-app-layout/test/app-layout.test.js

```diff
@@ -234,4 +237,81 @@ describe('vaadin-app-layout', () => {
       });
     });
   });
+
+  describe('notify children about resize', () => {
+    class ResizeAwareElement extends mixinBehaviors([IronResizableBehavior], PolymerElement) {
+      static get is() {
+        return 'resize-aware';
+      }
+    }
+
+    customElements.define(ResizeAwareElement.is, ResizeAwareElement);
+
+    let resizeAwareChild;
+
+    beforeEach(async () => {
+      layout = fixtureSync(`
+        <vaadin-app-layout>
+          <vaadin-drawer-toggle slot="navbar"></vaadin-drawer-toggle>
+          <h2 slot="navbar">App name</h2>
+          <section slot="drawer">
+            <p>Item 1</p>
+            <p>Item 2</p>
+            <p>Item 3</p>
+          </section>
+          <resize-aware></resize-aware>
+        </vaadin-app-layout>
+      `);
+      // force non-overlay mode as default
+      layout.style.setProperty('--vaadin-app-layout-drawer-overlay', 'false');
+      layout._updateOverlayMode();
+      await nextFrame();
+      resizeAwareChild = layout.querySelector('resize-aware');
+      sinon.stub(resizeAwareChild, 'notifyResize');
+    });
+    afterEach(() => {
+      resizeAwareChild.notifyResize.restore();
+    });
+
+    it('should notify when drawer opens or closes', () => {
+      layout.drawerOpened = false;
+      expect(resizeAwareChild.notifyResize.calledOnce).to.be.true;
+
+      resizeAwareChild.notifyResize.resetHistory();
+      layout.drawerOpened = true;
+      expect(resizeAwareChild.notifyResize.calledOnce).to.be.true;
+    });
+
+    it('should notify when drawer is hidden due to empty content', () => {
+      const section = layout.querySelector('[slot="drawer"]');
+      section.parentNode.removeChild(section);
+      layout._drawerChildObserver.flush();
+
+      expect(resizeAwareChild.notifyResize.calledOnce).to.be.true;
+    });
+
+    it('should notify when content is added to drawer', () => {
+      const newContent = document.createElement('div');
+      newContent.setAttribute('slot', 'drawer');
+      layout.appendChild(newContent);
+      layout._drawerChildObserver.flush();
+
+      expect(resizeAwareChild.notifyResize.calledOnce).to.be.true;
+    });
+
+    it('should notify when switching between regular and overlay mode', () => {
+      // force overlay mode
+      layout.style.setProperty('--vaadin-app-layout-drawer-overlay', 'true');
+      layout._updateOverlayMode();
+
+      expect(resizeAwareChild.notifyResize.calledOnce).to.be.true;
+
+      // remove overlay mode
+      resizeAwareChild.notifyResize.resetHistory();
+      layout.style.setProperty('--vaadin-app-layout-drawer-overlay', 'false');
+      layout._updateOverlayMode();
+
+      expect(resizeAwareChild.notifyResize.calledOnce).to.be.true;
```

Comment:
Considering [the presentation](https://docs.google.com/presentation/d/1-o-5p9N5mYUraNgC6eJaR63LySKli99Ba4g1ulMRBNU/edit#slide=id.gbed7c5ebbc_0_101), it could be a good candidate to be decoupled into two several suites.

Reply:
Since you commented in a test, do you mean a separate test? A test is everything in an `it` block, a test suite are all tests in a `describe` block.

Reply:
I was kind of misleading you in choosing this line and saying `test suite`. Sorry.

I meant this test contains **2 acts** and it probably tests **2 multiple things**:

1. Calling `notifyResize` when changing `--vaadin-app-layout-drawer-overlay` from `false` to `true`
2. Calling `notifyResize` when changing `--vaadin-app-layout-drawer-overlay` from `true` to `false`

That's why I thought it could be split into 2 tests considering the presentation, but I guess it requires creating another test suite also.

Reply:
I get the idea, but in this case I think it would hurt readability to split it up more. As you said, I would have to add another level of nesting by adding a new test suite where the overlay is already open, just to extract this single test. Besides there are existing tests which use this kind of toggling behaviour.

@web-padawan WDYT?

Reply:
I agree with @sissbruecker here, although there is definitely a room for improvement in the whole test suite here.
Some of the existing tests are already using this approach, and we probably don't want to change them all.

Let's keep this as a single test for now.